docs/in-depth-guide-scraper-functions.md

```diff
@@ -0,0 +1,319 @@
+[
+	IngredientGroup(
+		ingredients=[
+			'250g self-raising flour',
+			'25g cocoa powder',
+			'175g light muscovado sugar',
+			'85g unsalted butter , melted',
+			'5 tbsp vegetable or sunflower oil',
+			'150g pot fat-free natural yogurt',
+			'1 tsp vanilla extract', '3 large eggs'
+		],
+		purpose=None),
+	IngredientGroup(
+		ingredients=[
+			'85g unsalted butter , softened',
+			'1 tbsp milk',
+			'½ tsp vanilla extract',
+			'200g icing sugar , sifted',
+			'food colourings (optional)',
+			'sweets and sprinkles, to decorate'
+		],
+		purpose='For the frosting and decorating')
+]
+```
+
+
+
+### `instruction_list()`
+
+Return a `list` of instructions. This `list` is generated by splitting the output of `instructions()` on newline characters.
+
+```py
+>>> scraper.instructions_list()
+[
+	'Heat oven to 190C/170C fan/gas 5 and line a 12-hole muffin tin with deep cake cases. Put all the cake ingredients into a large bowl and beat together with electric hand beaters until smooth. Spoon the mix into the cases, then bake for 20 mins until risen and a skewer inserted into the middle comes out dry. Cool completely on a rack. Can be made up to 3 days ahead and kept in an airtight container, or frozen for up to 1 month.',
+ 	'For the frosting, work the butter, milk and vanilla into the icing sugar until creamy and pale. Colour with food colouring, if using, then create your own gruesome monster faces using sweets and sprinkles.'
+]
+```
+
+### `language() -> str`
+
+Returns the language of recipe page, as defined within the page's HTML.
+This is typically a two-letter BCP 47 language code, such as 'en' for English or 'de' for German,
+but may also include the dialect or variation, such as 'en-US' for American English.
+
+The language code is based on BCP 47 standards.
+For a comprehensive list of BCP 47 language codes, refer to this GitHub Gist:
+https://gist.github.com/typpo/b2b828a35e683b9bf8db91b5404f1bd1
+
+```py
+>>> scraper.language()
+'en'
+```
+
+### `links() -> List[Dict[str, str]]`
+
+Returns a `list` of all links found in the page HTML defined in an `<a>` element. For each link, the attributes of the HTML element are returned as a `dict`.
+
+```py
+>>> scraper.links()
+[
+	{
+		'class': ['popup-toggle-button'],
+		'aria-label': 'Toggle main-navigation popup',
+  		'aria-haspopup': 'true',
+  		'href': '#main-navigation-popup'
+	},
+	... # etc.
+]
+```
+
+### `site_name() -> str | None`
+
+Returns the website name, as defined in the page's HTML. If the page does not define this, this function returns None
+
+```py
+>>> scraper.site_name()
+None
+```
+
+### `to_json() -> List[str, str]`
+
+Returns the output of all functions implemented by this scraper as a `dict`.
+
+```py
+>>> scraper.to_json()
+{
+	'author': 'Good Food team',
+	'canonical_url': 'https://www.bbcgoodfood.com/recipes/monster-cupcakes',
+	'category': 'Treat',
+	... # etc.
+}
+```
+
+
+
+## Optional methods
+
+### `category() -> str`
+
+Returns the category of the recipe.
```

Comment:
```suggestion
Semi-structured field that can contain a mix of cuisine type (for example, country names), mealtime (breakfast/dinner/etc) and dietary properties (gluten-free, vegetarian).
```

Reply:
Generally agree, although the scrapers have a `cuisine()` function for cuisine information. That being said, I would expect `category()` to return whatever category the site defines, so that may well include a cuisine.